bokehjs/src/lib/models/widgets/tables/data_table.ts

```diff
@@ -34,6 +42,14 @@ export class TableDataProvider implements DataProvider<Item> {
 
   constructor(source: ColumnDataSource, view: CDSView) {
     this.init(source, view)
+    return new Proxy(this, {
+      get: (obj: any, key: string | number) => {
+        if (typeof(key) === 'string' && (Number.isInteger(Number(key)))) // key is an index
+          return obj.getItem(Number(key))
+        else
+          return obj[(key as any)]
+      }
+    })
```

Comment:
I thought you would hate this but I haven't found any other approach that would work. Basically for SlickGrid to correctly handle this DataProvider we have to make it appear like an Array and allow indexing.